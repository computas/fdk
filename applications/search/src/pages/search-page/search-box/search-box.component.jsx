import React from 'react';
import PropTypes from 'prop-types';
import { getParamFromUrl } from '../../../lib/addOrReplaceUrlParam';

import localization from '../../../lib/localization';
import { CustomHitsStats } from './custom-hits-stats/custom-hits-stats.component';
import './search-box.scss';

export const SearchBox = props => {
  const {
    onSearchSubmit,
    onSearchChange,
    searchQuery,
    countDatasets,
    countTerms,
    countApis,
    open
  } = props;
  let refSearchBox; // eslint-disable-line no-unused-vars
  return (
    <div className="row pt-5 pb-5">
      <div className="col-12 col-lg-8 offset-lg-2 fdk-search-flex">
        <div className="d-inline d-lg-none">
          <button
            type="button"
            className="fdk-button fdk-button-filter"
            onClick={open}
          >
            Filter
          </button>
        </div>
        <div className="fdk-search-box">
          <form
            onSubmit={e => {
              e.preventDefault();
              onSearchSubmit(e.target.value);
            }}
          >
            <label htmlFor="searchBox">
              <input
                name="searchBox"
                ref={input => {
                  refSearchBox = input;
                }}
                type="search"
                placeholder={localization.query.intro}
                aria-label={localization.query.intro}
                className="fdk-search"
                value={searchQuery || ''}
                onChange={e => {
                  e.preventDefault();
                  onSearchChange(e);
                }}
              />
            </label>
          </form>
        </div>
        <button
          type="button"
          onClick={refSearchBox => {
            onSearchSubmit(refSearchBox.target.value);
          }}
          className="fdk-button-search btn btn-lg"
        >
          <i className="fa fa-search mr-2" />
          {localization.query.do}
        </button>
      </div>
      <div className="col-md-12 text-center">
        <CustomHitsStats
          countDatasets={countDatasets}
          countTerms={countTerms}
          countApis={countApis}
          filteringOrTextSearchPerformed={
            !!(
              getParamFromUrl('q') ||
              getParamFromUrl('theme') ||
              getParamFromUrl('accessRight') ||
              getParamFromUrl('publisher')
            )
          }
        />
      </div>
    </div>
  );
};

SearchBox.defaultProps = {
  searchQuery: null,
  countDatasets: null,
  countTerms: null,
  countApis: null
};

SearchBox.propTypes = {
  onSearchSubmit: PropTypes.func.isRequired,
  onSearchChange: PropTypes.func.isRequired,
  searchQuery: PropTypes.string,
  countDatasets: PropTypes.number,
  countTerms: PropTypes.number,
  countApis: PropTypes.number,
  open: PropTypes.func.isRequired
};
