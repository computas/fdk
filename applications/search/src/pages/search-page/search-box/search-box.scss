@import '../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-search-flex {
  display: flex;
  align-items: flex-start;
  font-size: 2rem;

  .fdk-search-box {
    width: 100%;
    .glyphicon-search-frontpage {
      top: 20px;
    }
    label {
      width: 100%
    }
    input.fdk-search {
      color: $color1;
      border: none;
      border-radius: 0;
      box-shadow: 0 4px 8px rgba(45, 55, 65, 0.2);
      font-weight: 400;
      height: 54px;
      width: 100%;
      padding: 12px 16px;
      margin: 0 0 10px 0px;
    }
  }

  .fdk-button-filter {
    background-color: $color1;
    color: $color0;
    height: 54px;
    margin: 0;
    border-radius: 5px 0 0 5px;
    box-shadow: 0 4px 8px rgba(45, 55, 65, 0.2);
    &:hover, &:focus {
      color: $color0;
      border-color: $color1;
    }
  }

  .fdk-button-search {
    background-color: $color1;
    color: $color0;
    margin: 0;
    border-radius: 0 5px 5px 0;
    border-color: $color1;
    box-shadow: 0 4px 8px rgba(45, 55, 65, 0.2);
    height: 54px;
    &:hover, &:focus {
      color: $color0;
      border-color: $color1;
    }
  }
}

/* Medium Devices, Desktops */
@media only screen and (min-width : 992px) {

  .fdk-search-flex {
    .fdk-search-box {
      input.fdk-search {
        border-radius: 5px 0 0 5px;
        font-size: 3rem;
        height: 68px;
      }
    }

    .fdk-button-search {
      font-size: 2.5rem;
      height: 68px;
    }
  }

  .glyphicon-search-frontpage {
    border-radius: 5px;
    right: 26px ;
    top: 92px;
    font-size: 3rem;
  }

}

/* Large Devices, Wide Screens */
@media only screen and (min-width : 1200px) {

}