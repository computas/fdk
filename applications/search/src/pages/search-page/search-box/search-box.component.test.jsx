import React from 'react';
import { shallow } from 'enzyme';
import { SearchBox } from './search-box.component';

let onSearchSubmit;
let onSearchChange;
let preventDefault;
let open;
let defaultProps;

beforeEach(() => {
  onSearchSubmit = jest.fn();
  onSearchChange = jest.fn();
  open = jest.fn();
  preventDefault = jest.fn();
  defaultProps = {
    onSearchSubmit,
    onSearchChange,
    searchQuery: 'enhetsregister',
    countDatasets: 100,
    countTerms: 10,
    open
  };
});

test('should render SearchBox correctly', () => {
  const wrapper = shallow(<SearchBox {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should call onSearchSubmit prop for valid form submission', () => {
  const mockedEvent = {
    target: { value: 'test value' },
    preventDefault
  };
  const wrapper = shallow(<SearchBox {...defaultProps} />);
  wrapper.find('form').prop('onSubmit')(mockedEvent);
  expect(preventDefault).toHaveBeenCalled();
  expect(onSearchSubmit).toHaveBeenLastCalledWith(mockedEvent.target.value);
});

test('should call onSearchChange on change', () => {
  const mockedEvent = {
    target: { value: 'test value' },
    preventDefault
  };
  const wrapper = shallow(<SearchBox {...defaultProps} />);
  wrapper.find('input').prop('onChange')(mockedEvent);
  expect(preventDefault).toHaveBeenCalled();
  expect(onSearchChange).toHaveBeenLastCalledWith(mockedEvent);
});

test('should call onSearchSubmit prop for valid form submission', () => {
  const mockedEvent = {
    target: { value: 'test value' },
    preventDefault
  };
  const wrapper = shallow(<SearchBox {...defaultProps} />);
  wrapper.find('button.fdk-button-search').prop('onClick')(mockedEvent);
  expect(onSearchSubmit).toHaveBeenLastCalledWith(mockedEvent.target.value);
});

test('should call open when filter clicked', () => {
  const wrapper = shallow(<SearchBox {...defaultProps} />);
  wrapper.find('button.fdk-button-filter').prop('onClick')();
  expect(open).toHaveBeenCalled();
});
