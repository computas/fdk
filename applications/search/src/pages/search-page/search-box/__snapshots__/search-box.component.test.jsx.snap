// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBox correctly 1`] = `
<div
  className="row pt-5 pb-5"
>
  <div
    className="col-12 col-lg-8 offset-lg-2 fdk-search-flex"
  >
    <div
      className="d-inline d-lg-none"
    >
      <button
        className="fdk-button fdk-button-filter"
        onClick={[MockFunction]}
        type="button"
      >
        Filter
      </button>
    </div>
    <div
      className="fdk-search-box"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          htmlFor="searchBox"
        >
          <input
            aria-label="Search for datasets..."
            className="fdk-search"
            name="searchBox"
            onChange={[Function]}
            placeholder="Search for datasets..."
            type="search"
            value="enhetsregister"
          />
        </label>
      </form>
    </div>
    <button
      className="fdk-button-search btn btn-lg"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-search mr-2"
      />
      Search
    </button>
  </div>
  <div
    className="col-md-12 text-center"
  >
    <CustomHitsStats
      countApis={null}
      countDatasets={100}
      countTerms={10}
      filteringOrTextSearchPerformed={false}
    />
  </div>
</div>
`;
